pkg/apis/clickhouse-keeper.altinity.com/v1/type_spec.go

```diff
@@ -23,6 +23,7 @@ import (
 type ChkSpec struct {
 	TaskID                 *types.String       `json:"taskID,omitempty"                 yaml:"taskID,omitempty"`
 	NamespaceDomainPattern *types.String       `json:"namespaceDomainPattern,omitempty" yaml:"namespaceDomainPattern,omitempty"`
+	Suspend                *types.StringBool   `json:"suspend,omitempty"                yaml:"suspend,omitempty"`
```

Comment:
For pause/suspend option it would be nice to use annotations, like Cluster API project does
Pros: 
- No real object cast needed in predicate functions, just ObjectMeta extraction
- Common approach
- Transparent field migration

Reply:
Annotation-based implementation may be added as a second approach, both ways are possible and have pros/cons

Reply:
I am generally not a big fan of "annotations as an API", due to lack of schema validation and consistency.

There are many GitHub issues where kubernetes maintainers are also aligned on this. As mentioned above, there are pros/cons to each, but if you can drive it via a defined API, that's always best.

>https://github.com/kubernetes/kubernetes/issues/101776#issuecomment-841441066

"we should avoid using annotations as an API wherever possible"

>https://github.com/kubernetes/enhancements/issues/2255#issuecomment-1382598613

"I'm not in love with annotations as APIs. Do we REALLY think this is the best answer?"

Reply:
Ok, make sense, but sometimes annotations a more convenient option, when your work with unstructured or schemaless objects.
Agree with @sunsingerus to have both annotation and field based management is a good way.